content/help/ops/traffic-management/deploy-guidelines/index.md

```diff
@@ -6,6 +6,113 @@ weight: 5
+underlying microservices, will need to also be included in this single configuration file, instead of
+in separate resources associated with, and potentially owned by, the individual service teams.
+See [Route rules have no effect on ingress gateway requests](#route-rules-have-no-effect-on-ingress-gateway-requests)
+for details.
+
+To avoid this problem, it may be preferable to break up the configuration of `myapp.com` into several
+`VirtualService` fragments, one per implementation service. For example:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: myapp-service1
+spec:
+  hosts:
+  - myapp.com
+  gateways:
+  - myapp-gateway
+  http:
+  - match:
+    - uri:
+        prefix: /service1
+    route:
+    - destination:
+        host: service1.default.svc.cluster.local
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: myapp-service2
+spec:
+  hosts:
+  - myapp.com
+  gateways:
+  - myapp-gateway
+  http:
+  - match:
+    - uri:
+        prefix: /service2
+    route:
+    - destination:
+        host: service2.default.svc.cluster.local
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: myapp-...
+{{< /text >}}
+
+When a second and subsequent `VirtualService` for an existing host is applied, istio-pilot will merge
+the additional route rules into the existing configuration of the host. There are, however, several
+limitations with this feature that must be considered carefully when using it.
+
+1. Although the order of evaluation for rules in any given source `VirtualService` will be retained,
+   the cross-resource order is UNDEFINED. In other words, there is no guaranteed order of evaluation
+   for rules across the fragment configurations, so it will only have predictable behavior if there
+   are no conflicting rules or order dependency between rules across fragments.
+1. There should only be one "catch-all" rule (i.e., a rule that matches any request path or header) in the fragments.
+   All such "catch-all" rules will be moved to the end of the list in the merged configuration, but
+   since they are "catch-alls", whichever is applied first will essentially override and disable any others.
+1. A `VirtualService` can only be fragmented this way if it is bound to an ingress `Gateway`, i.e.,
+   host merging is currently only implemented by `istio-ingressgateway`. The feature is not supported in sidecars.
+
+A `DestinationRule` can also be fragmented with similar merge semantic and restrictions.
+
+1. The first definition of any given subset (e.g., `s1`) is used. Any following duplicates are discarded.
+   Therefore, there should not be more than one definition of subset `s1` across the fragments,
+   to make sure the same one will always be used.
+1. The first top-level `trafficPolicy` is used. Any following top-level `trafficPolicy` configuration is discarded.
```

Comment:
When top-level traffic policies are defined in multiple destination rules, Istio will pick the oldest rule with the top-level traffic policy.